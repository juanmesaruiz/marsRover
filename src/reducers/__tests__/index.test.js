import state from '../../store/__mocks__/mockState';

import reducer, { getGrid, getObstacles, getRover } from '../index';

describe('Reducer src/reducers/index.js', () => {
  it('Default', () => {
    const action = {};
    const initialState = undefined;
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  describe('Selectors:', () => {
    it('getGrid', () => {
      expect(getGrid(state)).toMatchSnapshot();
    });
    it('getObstacles', () => {
      expect(getObstacles(state)).toMatchSnapshot();
    });
    it('getRover', () => {
      expect(getRover(state)).toMatchSnapshot();
    });
  });
});
