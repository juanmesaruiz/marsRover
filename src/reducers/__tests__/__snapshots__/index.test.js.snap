// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer src/reducers/index.js Default 1`] = `
Object {
  "grid": Object {
    "x": 50,
    "y": 25,
  },
  "obstacles": Array [],
  "rover": Object {
    "current": null,
    "direction": "north",
    "log": Array [],
  },
}
`;

exports[`Reducer src/reducers/index.js Selectors: getGrid 1`] = `
Object {
  "x": 10,
  "y": 5,
}
`;

exports[`Reducer src/reducers/index.js Selectors: getObstacles 1`] = `
Array [
  Object {
    "x": 0,
    "y": 3,
  },
  Object {
    "x": 2,
    "y": 1,
  },
  Object {
    "x": 4,
    "y": 4,
  },
  Object {
    "x": 3,
    "y": 3,
  },
  Object {
    "x": 1,
    "y": 4,
  },
]
`;

exports[`Reducer src/reducers/index.js Selectors: getRover 1`] = `
Object {
  "current": Object {
    "x": 2,
    "y": 2,
  },
  "direction": "east",
  "log": Array [
    "Obstacle - X 0, Y 3",
    Object {
      "x": 0,
      "y": 1,
    },
    Object {
      "x": 0,
      "y": 0,
    },
  ],
}
`;
